tests/cross/src/test/scala/tests/pc/CompletionWorkspaceSuite.scala

```diff
@@ -344,21 +344,11 @@ class CompletionWorkspaceSuite extends BaseCompletionSuite {
     """|import java.util.ArrayDeque
        |object Main {
        |  def foo(): Unit = null match {
-       |    case x: ArrayDeque[$0] =>
```

Comment:
ArrayDeque takes type parameters. doesn't it? So why is this removed?

Reply:
The current one doesn't compile. It should have type parameters.

Reply:
I would turn into an issue after merge

I don't know why it's completed with `[]` in the main branch. 
There is no handler that marks this position by `isTypePosition` for the tree in main  or this case here: 
https://github.com/scalameta/metals/blob/c328f0925e9692b493aba395492ae03fd36da424/mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala#L121-L152
While the tree is: `Ident(ArrayDeque), Typed(Ident(_),Ident(ArrayDeque)), Bind(x,Typed(Ident(_),Ident(ArrayDeque))), CaseDef(Bind(x,Typed(Ident(_),Ident(ArrayDeque)))`

Reply:
Perhaps, this would lead to not adding square brackets also in other positions. Also, what about the other PR with substantial changes that would not be usable after this merge? Also, this is at the last minute, what if it introduces other problems? The regression can be solved by simply adding `|| sym.isTerm` in `isNotClassOrTraitOrTheyAreValidForPos` 
What if the changes of the other PR cannot be reworked on this one?

Reply:
yep, `isTerm` fixes this case but we also need to deal with objects.
I've checked that tests doesn't pass in your PR.